connections.go

```diff
@@ -0,0 +1,245 @@
+package packngo
+
+import (
+	"path"
+)
+
+type ConnectionRedundancy string
+type ConnectionType string
+
+const (
+	connectionBasePath                           = "/connections"
+	virtualCircuitsBasePath                      = "/virtual-circuits"
+	ConnectionShared        ConnectionType       = "shared"
+	ConnectionDedicated     ConnectionType       = "dedicated"
+	ConnectionRedundant     ConnectionRedundancy = "redundant"
+	ConnectionPrimary       ConnectionRedundancy = "primary"
+)
+
+type ConnectionService interface {
+	OrganizationCreate(string, *ConnectionCreateRequest) (*Connection, *Response, error)
+	ProjectCreate(string, *ConnectionCreateRequest) (*Connection, *Response, error)
+	OrganizationList(string, *GetOptions) ([]Connection, *Response, error)
+	ProjectList(string, *GetOptions) ([]Connection, *Response, error)
+	Delete(string) (*Response, error)
+	Get(string, *GetOptions) (*Connection, *Response, error)
+	Events(string, *GetOptions) ([]Event, *Response, error)
+	PortEvents(string, string, *GetOptions) ([]Event, *Response, error)
+	VirtualCircuitEvents(string, *GetOptions) ([]Event, *Response, error)
+	Ports(string, *GetOptions) ([]ConnectionPort, *Response, error)
+	Port(string, string, *GetOptions) (*ConnectionPort, *Response, error)
+	VirtualCircuits(string, string, *GetOptions) ([]ConnectionVirtualCircuit, *Response, error)
+	VirtualCircuit(string, *GetOptions) (*ConnectionVirtualCircuit, *Response, error)
+	DeleteVirtualCircuit(string) (*Response, error)
+}
+
+type ConnectionServiceOp struct {
+	client *Client
+}
+
+type connectionPortsRoot struct {
+	Ports []ConnectionPort `json:"ports"`
+}
+
+type virtualCircuitsRoot struct {
+	VirtualCircuits []ConnectionVirtualCircuit `json:"virtual_circuits"`
+	Meta            meta                       `json:"meta"`
+}
+
+type connectionsRoot struct {
+	Connections []Connection `json:"interconnections"`
+	Meta        meta         `json:"meta"`
+}
+
+type ConnectionVirtualCircuit struct {
+	ID      string          `json:"id"`
+	Name    string          `json:"name,omitempty"`
+	Status  string          `json:"status,omitempty"`
+	VNID    string          `json:"vnid,omitempty"`
+	NniVNID string          `json:"nni_vnid,omitempty"`
+	NniVLAN string          `json:"nni_vlan,omitempty"`
+	Project *Project        `json:"project,omitempty"`
+	Port    *ConnectionPort `json:"port,omitempty"`
+}
+
+type ConnectionPort struct {
+	ID              string                     `json:"id"`
+	Name            string                     `json:"name,omitempty"`
+	Status          string                     `json:"status,omitempty"`
+	Role            string                     `json:"role,omitempty"`
+	Speed           string                     `json:"speed,omitempty"`
+	Organization    *Organization              `json:"organization,omitempty"`
+	VirtualCircuits []ConnectionVirtualCircuit `json:"virtual_circuits,omitempty"`
+	LinkStatus      string                     `json:"link_status,omitempty"`
```

Comment:
Do we want to create an enum for LinkStatus too? (I didn't look into what values are possible)

We haven't done much of this to this point, but I'm not opposed to introducing enums in new resources (or backfilling these for known enums).

Reply:
Same comment would apply to any Status field in this PR: https://github.com/packethost/packngo/pull/234/files#diff-e772df221260f0ad6138a54d148e7ec9af6cd92a8ff527a1a376171e1c27f83fR80

What do you think, @t0mk ?

Reply:
@displague it defintely makes sense to have enum there, but I would be guessing the possible values.

I checked the api-docs json for the metadata for interconnections, and the status is there just as string without enumerated values specified.

If you know where to look in the EqMetal internal codebase, please let me know and I'll make it enum.

Reply:
I dug up some possible values, but it would be best to have them included in the API spec or at least verified. (@jordan0day)

Reply:
Good call, I will add a ticket to get these values added to the api spec